x/lightclient/keeper/rollback.go

```diff
@@ -2,24 +2,117 @@ package keeper
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/ibc-go/v7/modules/core/exported"
+	"github.com/dymensionxyz/dymension/v3/x/lightclient/types"
+
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	commitmenttypes "github.com/cosmos/ibc-go/v7/modules/core/23-commitment/types"
+
+	ibctm "github.com/cosmos/ibc-go/v7/modules/light-clients/07-tendermint"
 )
 
-func (hook rollappHook) OnHardFork(sdk.Context, string, uint64) error {
-	/*
-		// freeze IBC client state
-		clientState, ok := k.ibcClientKeeper.GetClientState(ctx, clientId)
-		if !ok {
-			return errorsmod.Wrapf(types.ErrInvalidClientState, "client state for clientID %s not found", clientId)
-		}
+func (hook rollappHook) OnHardFork(ctx sdk.Context, rollappId string, height uint64) error {
+	hook.k.RollbackCanonicalClient(ctx, rollappId, height)
+	return nil
+}
+
+func (k Keeper) RollbackCanonicalClient(ctx sdk.Context, rollappId string, height uint64) {
+	client, found := k.GetCanonicalClient(ctx, rollappId)
+	if !found {
+		k.Logger(ctx).Error("Canonical client not found", "rollappId", rollappId)
+		return
+	}
+	cs := k.ibcClientKeeper.ClientStore(ctx, client)
 
-		tmClientState, ok := clientState.(*cometbfttypes.ClientState)
-		if !ok {
-			return errorsmod.Wrapf(types.ErrInvalidClientState, "client state with ID %s is not a tendermint client state", clientId)
+	// iterate over all consensus states and metadata in the client store
+	ibctm.IterateConsensusStateAscending(cs, func(h exported.Height) bool {
+		// if the height is lower than the target height, continue
+		if h.GetRevisionHeight() < height {
+			return false
 		}
```

Comment:
2 things here:
1. if it's target height it should probably be  `<=` (unless it's the fraud height?)
2. the acennding iteration could be very pricy. desc iteration would be far more efficeint but I guess we don't have this api? thinking of 1 year of iteration.. not sure if the consensus states get pruned at some point?

Reply:
1. it's ` fraud height`
2. consensus states got pruned after `TrustingPeriod`. anyway I'll change it to descending